Source/DafnyCore/Resolver/PreType/TypeAdjustments.cs

```diff
@@ -0,0 +1,274 @@
+//-----------------------------------------------------------------------------
```

Comment:
Is this code mostly utilities for the TypeAdjustor?

Reply:
The code contains utilities for the PreType-to-Type phase. (That hadn't been clear to me when I started writing it.)

I split this file into two:
* `AdjustableType.cs` now contains the declaration of the class `AdjustableType` and the two `...TypePlaceholder` classes (which may change before this work is done)
* `PreType2TypeUtil.cs` contains class `TypeAdjustments` (which I have renamed `PreType2TypeUtil` and made into a `static` class)